respx/models.py

```diff
@@ -252,26 +254,39 @@ def get_url(self) -> Optional[URLPatternTypes]:
         return self._url
 
     def set_url(
-        self, url: Optional[URLPatternTypes], base: Optional[str] = None
+        self,
+        url: Optional[URLPatternTypes],
+        base: Optional[str] = None,
+        params: Optional[QueryParamTypes] = None,
     ) -> None:
+        params = str(httpx.QueryParams(params))
```

Comment:
Since we start relying on a more stable and public HTTPX model here, I think we should go even further and start using `httpx.URL` internally.

How about constructing and setting `self._url` as an `httpx.URL` instance in `set_url`?

We still need to support regexp though, and handle it as is. I don't think we need to support joining regexp-urls with params, but we do support base_url + regexp, so if easy to extend the regexp with params then great, and if not, then we should raise when passing regex + params.

Pattern matching needs some alignment as well I think if we move to `httpx.URL`.